src/compiler/core.ts

```diff
@@ -496,18 +505,25 @@ namespace ts {
      * @param array A sorted array whose first element must be no larger than number
      * @param number The value to be searched for in the array.
      */
-    export function binarySearch(array: number[], value: number): number {
+    export function binarySearch<T>(array: T[], value: T, comparer?: (v1: T, v2: T) => number): number {
+        if (!array || array.length === 0) {
+            return -1;
+        }
+
         let low = 0;
         let high = array.length - 1;
+        comparer = comparer !== undefined

```

Comment:
I understand this is not part of this change. but can we replace this with a check at the use site, and other wise use the normal loop. optionally have two implementations. it would not make sens to have a function call just to return `>`